### PR TITLE
PIM-10559: [Backport PIM-10478]: Disable compute_family_variant_structure_changes on familySave

### PR DESCRIPTION
**Pull request by cemorin**
**Description (for Contributor and Core Developer)**

Backport of this PR: https://github.com/akeneo/pim-community-dev/pull/17298

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: tests/back/Pim/Structure/Specification/Bundle/EventSubscriber/SaveFamilyVariantOnFamilyUpdateSubscriberSpec.php [hunk 1]
```diff
@@ -245,4 +230,31 @@ function it_does_not_bulk_save_if_on_non_unitary_save_and_POST_SAVE_ALL_event(
 
         $this->onBulkSave($event);
     }
+
+    function it_does_not_force_launch_compute_family_variant_job_when_attributes_did_not_change(
+        ValidatorInterface $validator,
+        BulkSaverInterface $bulkFamilyVariantSaver,
+        GenericEvent $event,
+        ConstraintViolationList $constraintViolationList,
```
Comment (mmetayer) [inline tests/back/Pim/Structure/Specification/Bundle/EventSubscriber/SaveFamilyVariantOnFamilyUpdateSubscriberSpec.php:238]:
```suggestion
        ConstraintViolationList $constraintViolationList
```
php 7.4 :slightly_smiling_face:

File: tests/back/Pim/Structure/Specification/Component/Model/FamilySpec.php [hunk 2]
```diff
@@ -57,4 +58,43 @@ function it_allows_defining_an_attribute_to_use_as_label(AttributeInterface $nam
         $this->getAttributeAsLabel()->shouldReturn(null);
         $this->setAttributeAsLabel($name)->shouldReturn($this);
     }
+
+    function it_updates_a_family_by_adding_an_attribute(
+        AttributeInterface $skuAttribute,
+        AttributeInterface $nameAttribute,
+        AttributeInterface $descAttribute,
```
Comment (mmetayer) [inline tests/back/Pim/Structure/Specification/Component/Model/FamilySpec.php:65]:
```suggestion
        AttributeInterface $descAttribute
```

File: tests/back/Pim/Structure/Specification/Component/Model/FamilySpec.php [hunk 3]
```diff
@@ -57,4 +58,43 @@ function it_allows_defining_an_attribute_to_use_as_label(AttributeInterface $nam
...
+
+        $this->updateAttributes([$skuAttribute, $nameAttribute, $descAttribute]);
+
+        $this->getAttributeCodes()->shouldReturn(['sku', 'name', 'description']);
+    }
+
+    function it_updates_a_family_by_removing_an_attribute(
+        AttributeInterface $skuAttribute,
+        AttributeInterface $nameAttribute,
+        AttributeInterface $descAttribute,
```
Comment (mmetayer) [inline tests/back/Pim/Structure/Specification/Component/Model/FamilySpec.php:84]:
```suggestion
        AttributeInterface $descAttribute
```

File: tests/back/Pim/Structure/Specification/Component/Updater/FamilyUpdaterSpec.php [hunk 4]
```diff
@@ -32,15 +32,18 @@ function let(
         ChannelRepositoryInterface $channelRepository,
         AttributeRequirementFactory $attrRequiFactory,
         AttributeRequirementRepositoryInterface $attributeRequirementRepo,
-        TranslatableUpdater $translatableUpdater
+        TranslatableUpdater $translatableUpdater,
+        IdentifiableObjectRepositoryInterface $localeRepository,
```
Comment (mmetayer) [inline tests/back/Pim/Structure/Specification/Component/Updater/FamilyUpdaterSpec.php:36]:
```suggestion
        IdentifiableObjectRepositoryInterface $localeRepository
```

**Review by mmetayer — APPROVED** [review]

**Review by SamirBoulil — APPROVED** [review]

**Pull request merged**